### DISCUSSION TITLE
Aidbox Notebooks. REST & SQL console

### DISCUSSION THREAD

**Discussion by KGOH** [Aidbox/Issues#412] [Show and tell]
## Feature description
Aidbox Notebooks is an interactive environment that you can use to create and share documents consisting of different API calls.

![image](https://user-images.githubusercontent.com/11963737/128494095-db34533e-b682-4620-91c3-a0964585b140.png)

### Features:
- [x] Notebook save
- [x] Notebooks repository
- [x] Notebook export/import
- [ ] Cell export/import

### Modes:
- [x] REST
- [x] SQL
- [x] Markdown
- [ ] GraphQL
- [ ] JavaScript
- [ ] [Stresty](https://github.com/HealthSamurai/stresty)
- [ ] Different visualizations

**Comment by IvannikovG** [comment]
User experience might be enhanced by implementing these points (or some of them):

1). Highlighting of the current notebook is enabled (The notebook with active cursor)

2). More contrast or bigger button to run the notebook

3). Big notebook response (e.g. sql with 1000 json entries) is partly hidden by default and opened when asked explicitly

4). Html title of the page (tab name) is confusing: might be better if it has the same name as the notebook.

5). Page autoscrolling if the new notebook cell is in the bottom of the page (it goes down the page and not seen)

6). Some autosave shall be done. (I have made an error and 1 hour of work was destroyed, very crucial!). May be autosave when any 
cell is run.

7).  If one runs the cell and it results in error, then one fixes the error and runs the cell again - strange ui behaviour happens (see the link: https://ibb.co/V324G7w)

8). (nice to have) Drag and drop notebooks

9). Notebook while running should show that it is running

10). I found no way to delete notebook.

A lot of space is unused o the right of the page - potential good place to configure the environment for the notebooks (change bg colour/ set variables(?) etc/ ....)

Notebooks are TOP!

**Reply by Nesmeshnoy** [reply]
@IvannikovG, thank you for the input.

6 and 9 done.

**Comment by almartel** [comment]
This is a great feature - I've just started to explore it!
I would like to be able to clear the variables etc - similar to kernel reset in jupyter. I think I am getting strange behavior when I rerun cells. Some functionality to run all cells again or run all the cells above the selected would also be great. 
There also seems to be a timing issue - if I try running a cell that takes the output from the previous one, it sometimes fails on the first try, and succeeds on the second try.
